ext/curl/multi.c

```diff
@@ -76,17 +86,12 @@ PHP_FUNCTION(curl_multi_add_handle)
 	CURLMcode error = CURLM_OK;
 
 	ZEND_PARSE_PARAMETERS_START(2,2)
-		Z_PARAM_RESOURCE(z_mh)
-		Z_PARAM_RESOURCE(z_ch)
+		Z_PARAM_OBJECT_OF_CLASS(z_mh, curl_multi_ce)
+		Z_PARAM_OBJECT_OF_CLASS(z_ch, curl_ce)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if ((mh = (php_curlm *)zend_fetch_resource(Z_RES_P(z_mh), le_curl_multi_handle_name, le_curl_multi_handle)) == NULL) {
-		RETURN_THROWS();
-	}
-
-	if ((ch = (php_curl *)zend_fetch_resource(Z_RES_P(z_ch), le_curl_name, le_curl)) == NULL) {
-		RETURN_THROWS();
-	}
+	mh = Z_CURL_MULTI_P(z_mh);
+	ch = Z_CURL_P(z_ch);
```

Comment:
The GC_ADDREF(Z_RES_P(z_ch)) should be GC_ADDREF(Z_OBJ_P(z_ch)) below (also in one more place). Or just Z_ADDREF_P(z_ch) actually.